test/src/userobjects/InterfaceDomainUserObject.C

```diff
@@ -0,0 +1,164 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "InterfaceDomainUserObject.h"
+#include "MooseMesh.h"
+#include "Function.h"
+#include "MooseVariableFE.h"
+#include "Assembly.h"
+
+registerMooseObject("MooseTestApp", InterfaceDomainUserObject);
+
+InputParameters
+InterfaceDomainUserObject::validParams()
+{
+  InputParameters params = DomainUserObject::validParams();
+  params.addRequiredCoupledVar("u", "The variable we are diffusion for");
+  params.addRequiredCoupledVar("v", "The variable on the other side of the interface");
+  params.addParam<FunctionName>("function", "1", "The forcing function.");
+  params.addParam<Real>("robin_coef", 2, "The cofficient multiplying the Robin boundary condition");
+  params.addParam<std::vector<BoundaryName>>(
+      "robin_boundaries", "The boundaries on which to apply the Robin boundary condition");
+  params.addParam<std::vector<BoundaryName>>(
+      "interface_boundaries",
+      "The interface boundaries on which to apply the continuity of flux and value conditions");
+  params.addRequiredParam<Real>(
+      "interface_penalty",
+      "The penalty that penalizes jump between primary and neighbor variables.");
+  params.addRequiredParam<Real>("nl_abs_tol", "The absolute tolerance of the nonlinear solver");
+  return params;
+}
+
+InterfaceDomainUserObject::InterfaceDomainUserObject(const InputParameters & parameters)
+  : DomainUserObject(parameters),
+    _u(coupledValue("u")),
+    _v_neighbor(coupledNeighborValue("v")),
+    _grad_u(coupledGradient("u")),
+    _grad_v_neighbor(*coupledNeighborGradients("v")[0]),
+    _var(*getVar("u", 0)),
+    _v_var(*getVar("v", 0)),
```

Comment:
Can we create a function `getInterfaceVar`, which calls `getVar` and push the variable into a list that will be used in `checkVariable`?

Reply:
Potentially